default-token-account/program/tests/functional.rs

```diff
@@ -0,0 +1,349 @@
+use solana_program::{
+    instruction::InstructionError, program_pack::Pack, pubkey::Pubkey, system_instruction,
+    sysvar::rent::Rent,
+};
+use solana_program_test::*;
+use solana_sdk::{
+    signature::Signer,
+    transaction::{Transaction, TransactionError},
+};
+use spl_default_token_account::{
+    error::DefaultTokenAccountError, get_default_token_account_address, id, instruction, processor,
+};
+
+fn program_test() -> ProgramTest {
+    let mut pc = ProgramTest::new();
+
+    // Add Default Token Account program
+    pc.add_bpf_program(
+        "spl_default_token_account",
+        id(),
+        processor!(processor::process_instruction),
+    );
+
+    // Add Token program
+    pc.add_bpf_program(
+        "spl_token",
+        spl_token::id(),
+        processor!(spl_token::processor::Processor::process),
+    );
+
+    // Dial down the BPF compute budget to detect if the program gets bloated in the future
+    pc.set_bpf_compute_max_units(25_000);
```

Comment:
Demo of how to https://github.com/solana-labs/solana/pull/13245 is employed to in a `solana-program-test`-based test harness